src/main/java/org/spongepowered/api/map/font/MapFont.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Optional;
+
+/**
+ * Represents a font on a map that can be used for drawing characters.
+ */
+public final class MapFont {
+
+    private final ImmutableMap<Character, CharacterSprite> characters;
+
+    public MapFont(Map<Character, CharacterSprite> characters, int spaceSize) {
+        ImmutableMap.Builder<Character, CharacterSprite> builder =
+                new ImmutableMap.Builder<Character, CharacterSprite>();
+
+        if (!characters.containsKey(' ')) {
+            builder.put(' ', new CharacterSprite(spaceSize, 8, new boolean[16]));
+        }
+
+        builder.putAll(characters);
+
+        this.characters = builder.build();
+    }
+
+    /**
+     * Returns the {@link CharacterSprite} data for the specified character.
+     *
+     * @param ch The character to retrieve a sprite for
+     * @return The sprite version of the character
+     */
+    public Optional<CharacterSprite> getChar(char ch) {
+        return Optional.ofNullable(characters.get(ch));
+    }
+
+    /**
+     * Returns the pixel width of a string of text accounting for the
+     * {@link CharacterSprite} sizes and a spacing between characters.
+     *
+     * @param text The string of text to analyze the width of
+     * @return The width of the text in pixels
+     * @throws IllegalArgumentException If the text provided contains a
+     *         character not provided in the character sprites
+     */
+    public int getWidth(String text) {
+        checkNotNull(text, "text");
+
+        if (text.length() == 0) {
+            return 0;
+        }
+
+        int width = 0;
+        // Avoids copying the inner character array by indexing
+        for (int i = 0; i < text.length(); i++) {
+            char ch = text.charAt(i);
+            CharacterSprite value = characters.get(ch);
+            if (value == null) {
+                throw new IllegalArgumentException("Character '" + ch +
+                        "' was not provided as a CharacterSprite.");
+            }
+            width += value.getWidth();
+
+            // Only add spacing from the left after first character.
+            if (i > 1) {
+                width += value.getSpacing();
+            }
+        }
+        return width;
+    }
+
+    /**
+     * Returns the maximum pixel height of a string of text.
+     *
+     * @param text The text to return the height of
+     * @return The maximum number of pixels in height this text will take up
+     */
+    public int getHeight(String text) {
+        checkNotNull(text, "text");
+
+        if (text.length() == 0) {
+            return 0;
+        }
+
+        int maxHeight = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char ch = text.charAt(i);
+            // Skip whitespace
+            if (!Character.isWhitespace(ch)) {
+                CharacterSprite sprite = characters.get(ch);
+                if (sprite == null) {
+                    throw new IllegalArgumentException("Charcter '" + ch +
```

Comment:
`checkArgument(sprite != null, "Character '" + ch + "' was not provided as a CharacterSprite.");`